add example for extended patches (#1388)
kubernetes-sigs/kustomize: examples/patchMultipleObjects.md

[Strategic Merge Patch]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md
[JSON patches]: https://tools.ietf.org/html/rfc6902
[label selector]: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors

# Demo: applying a patch to multiple resources

A kustomization file supports customizing resources via both
[Strategic Merge Patch] and [JSON patches]. Now one patch can be
applied to multiple resources.

> The format of doing this is by specifying a patch and a target selector.

Reviewer: ```suggestion
This can be done by specifying a patch and a target selector as follows:
```

Author: done

> It is as the following:

Reviewer: ```suggestion
```

Author: done

    spec:
      containers:
        - name: nginx
          image: nginx
          args:
          - one
          - two
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: deploy2
spec:
  template:
    metadata:
      labels:
        key: value
    spec:
      containers:
        - name: busybox
          image: busybox
EOF
```

Declare a Strategic Merge Patch file to inject a sidecar container:

<!-- @addPatch @test -->
```
cat <<EOF >$DEMO_HOME/patch.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: not-important
spec:
  template:
    spec:
      containers:
        - name: istio-proxy
          image: docker.io/istio/proxyv2
          args:
          - proxy
          - sidecar
EOF
```

Apply the patch by adding _patches_ field in kustomization.yaml

<!-- @applyPatch @test -->
```
cat <<EOF >>$DEMO_HOME/kustomization.yaml
patches:
- path: patch.yaml
  target:
    kind: Deployment
EOF
```

Running `kustomize build $DEMO_HOME`, in the output confirm that both Deployment resources are patched correctly.

<!-- @confirmPatch @test -->
```
test 2 == \
  $(kustomize build $DEMO_HOME | grep "image: docker.io/istio/proxyv2" | wc -l); \
  echo $?

> ```

Reviewer: Feel free to also include the actual output of the patch for completeness, if you think it makes sense to do so.

Author: will add it.

Author: done
